source/plugins/go/src/oms.go

```diff
@@ -1059,26 +1061,44 @@ func PostTelegrafMetricsToLA(telegrafRecords []map[interface{}]interface{}) int
 				}
 			}
 			msgpBytes := convertMsgPackEntriesToMsgpBytes(MdsdInsightsMetricsTagName, msgPackEntries)
-			if MdsdInsightsMetricsMsgpUnixSocketClient == nil {
-				Log("Error::mdsd::mdsd connection does not exist. re-connecting ...")
-				CreateMDSDClient(InsightsMetrics, ContainerType)
+			var bts int
+			var er error
+			if IsWindows == false {
 				if MdsdInsightsMetricsMsgpUnixSocketClient == nil {
-					Log("Error::mdsd::Unable to create mdsd client for insights metrics. Please check error log.")
-					ContainerLogTelemetryMutex.Lock()
-					defer ContainerLogTelemetryMutex.Unlock()
-					InsightsMetricsMDSDClientCreateErrors += 1
-					return output.FLB_RETRY
+					Log("Error::mdsd::mdsd connection does not exist. re-connecting ...")
+					CreateMDSDClient(InsightsMetrics, ContainerType)
+					if MdsdInsightsMetricsMsgpUnixSocketClient == nil {
+						Log("Error::mdsd::Unable to create mdsd client for insights metrics. Please check error log.")
+						ContainerLogTelemetryMutex.Lock()
+						defer ContainerLogTelemetryMutex.Unlock()
+						InsightsMetricsMDSDClientCreateErrors += 1
+						return output.FLB_RETRY
+					}
 				}
-			}
 
-			deadline := 10 * time.Second
-			MdsdInsightsMetricsMsgpUnixSocketClient.SetWriteDeadline(time.Now().Add(deadline)) //this is based of clock time, so cannot reuse
-			bts, er := MdsdInsightsMetricsMsgpUnixSocketClient.Write(msgpBytes)
+				deadline := 10 * time.Second
+				MdsdInsightsMetricsMsgpUnixSocketClient.SetWriteDeadline(time.Now().Add(deadline)) //this is based of clock time, so cannot reuse
+				bts, er = MdsdInsightsMetricsMsgpUnixSocketClient.Write(msgpBytes)
+			} else {
+				if InsightsMetricsNamedPipe == nil {
+					EnsureGenevaOr3PNamedPipeExists(&InsightsMetricsNamedPipe, InsightsMetricsDataType, &ContainerLogsWindowsAMAClientCreateErrors, false, &MdsdContainerLogTagRefreshTracker)
+					if InsightsMetricsNamedPipe == nil {
+						Log("Error::mdsd::Unable to create mdsd client for insights metrics. Please check error log.")
+						ContainerLogTelemetryMutex.Lock()
+						defer ContainerLogTelemetryMutex.Unlock()
+						InsightsMetricsMDSDClientCreateErrors += 1
```

Comment:
we should track telemetry separately for the windows instead of tracking with InsightsMetricsMDSDClientCreateErrors which is specific to Linux .